protocols/gossipsub/src/behaviour.rs

```diff
@@ -2867,146 +2809,21 @@ where
         self.pending_iwant_msgs.clear();
     }
 
-    /// Send a [`Rpc`] message to a peer. This will wrap the message in an arc if it
+    /// Send a [`RpcOut`] message to a peer. This will wrap the message in an arc if it
     /// is not already an arc.
-    fn send_message(&mut self, peer_id: PeerId, message: proto::RPC) -> Result<(), PublishError> {
+    fn send_message(&mut self, peer_id: PeerId, message: RpcOut) -> Result<(), PublishError> {
```

Comment:
Perhaps not in this PR but the metrics and logging of what we are doing should be moved to this function in order to de-duplicate it from all the call-sites.

Reply:
did it on an isolated commit: fdee626aa873fea736b2e6dde05c22ebc4e14147